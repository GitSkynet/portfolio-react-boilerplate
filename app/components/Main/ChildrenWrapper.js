import styled from 'styled-components';
import { BASE_LINE_HEIGHT } from '../../styles/typography';

const ChildrenWrapper = styled.div`
  height: 100%;
  width: 100%;

  @supports (display: grid) {
    display: grid;
    grid-area: content;
  }

  & * + * {
    margin-top: ${BASE_LINE_HEIGHT}rem;
  }

  & * + *:not(p) {
    margin: ${BASE_LINE_HEIGHT * 2}rem 0;
  }

  & * + h2,
  & * + h3 {
    /* em, not rem, now */
    margin-top: ${BASE_LINE_HEIGHT}em;
  }
`;

export default ChildrenWrapper;
