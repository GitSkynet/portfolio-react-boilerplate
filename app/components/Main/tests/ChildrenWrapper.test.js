import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import ChildrenWrapper from '../ChildrenWrapper';

describe('<ChildrenWrapper />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(<ChildrenWrapper />);
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
