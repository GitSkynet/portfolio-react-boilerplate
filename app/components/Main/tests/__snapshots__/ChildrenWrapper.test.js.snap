// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChildrenWrapper /> should render and match the snapshot 1`] = `
.c0 {
  height: 100%;
  width: 100%;
}

.c0 * + * {
  margin-top: 1.5rem;
}

.c0 * + *:not(p) {
  margin: 3rem 0;
}

.c0 * + h2,
.c0 * + h3 {
  margin-top: 1.5em;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-area: content;
  }
}

<div
  className="c0"
/>
`;
